locallibrary/catalog/forms.py

```diff
@@ -0,0 +1,20 @@
+import datetime
+
+from django import forms
+from django.utils.translation import gettext
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(help_text=gettext('Enter a date between now and 4 weeks (default 3).'))
+
+    def clean_renewal_date(self):
+        data = self.cleaned_data['renewal_date']
```

Comment:
`cleaned_data` -> thuộc tính này ở đâu ra em? lưu những thông tin gì?

Reply:
cleaned_data : làm sạch dữ liệu và loại bỏ những cái nhập vào không an toàn từ người dùng
lưu thông tin ngày trả sách mà người dùng vừa nhập vào